src/wip/fixed-grid.rain

```diff
@@ -0,0 +1,660 @@
+
+#test-current-time !Test current time for plotting.
+#test-last-time !Test last time for plotting.
+
+#ensure-cooldown !Binding to enforce cooldown check.
+#ensure-grid-band !Binding to enforce price remains between minimum and maximum price set.
+#calculate-exact-quote !Binding to calculate exact amount of distribution tokens required for buys and sells.
+#calculate-io-ratio !Binding to calculate io-ratio for the order.
+#calculate-twap-ratio !Binding to calculate order to 30min twap ratio.
+#enforce-twap-check !Binding to enforce oracle manipulation protection guard.
+#io-multiplier !Binding to get the ratio multiplier for distribution token.
+#plottables !The binding for additional things we want to plot during testing.
+
+#uniswap-words !The subparser for the Uniswap words
+#orderbook-subparser !The subparser for the Orderbook words
+#cooldown-key "cooldown-key"
+
+#enforce-twap-check-chart
+  _: ;
+
+#enforce-twap-check-prod
+  twap-ratio: ,
+  :ensure(
+      less-than(
+        twap-ratio
+        add(1 twap-deviation)
+        )
+    "twap check"
+  );
+
+#calculate-twap-ratio-buy
+  io-ratio ratio-multiplier: ,
+  twap-30: mul(
+    uniswap-v3-twap-output-ratio(reserve-token distribution-token 1800 0 twap-io-fee)
+    ratio-multiplier
+  ),
+  twap-ratio: div(twap-30 io-ratio),
+  :call<'enforce-twap-check>(twap-ratio);
+
+#calculate-twap-ratio-sell
+  io-ratio ratio-multiplier: ,
+  twap-30: mul(
+    uniswap-v3-twap-output-ratio(distribution-token reserve-token 1800 0 twap-io-fee)
+    ratio-multiplier
+  ),
+  twap-ratio: div(twap-30 io-ratio),
+  :call<'enforce-twap-check>(twap-ratio);
+
+#jittery-binomial
+  seed:,
+  binomial18-10:
+    bitwise-count-ones(bitwise-decode<0 jittery-binomial-bits>(hash(seed))),
+  noise18-1:
+    mod(hash(seed 0) 1),
+  jittery-combined:
+    add(binomial18-10 noise18-1),
+  jittery-combined-normalized:
+    div(jittery-combined add(1 jittery-binomial-bits));
+
+#bounty-auction
+  time-since-cooldown: ,
+  bounty: min(
+    linear-growth(
+      bounty-min
+      div(bounty-unit-increase bounty-unit-time)
+      time-since-cooldown
+    )
+    bounty-max
+  );
+
+#ensure-cooldown-metric
+ _ _: block-timestamp() 0;
+
+#ensure-cooldown-chart
+ _ _: test-current-time test-last-time;
+ 
+#ensure-cooldown-prod
+ current-time: block-timestamp(),
+ last-time: get(hash(order-hash() cooldown-key)),
+ :ensure(
+   greater-than-or-equal-to(
+     current-time
+     add(last-time cooldown-time)
+   )
+   "cooldown"
+ ),
+ :set(hash(order-hash() cooldown-key) current-time);
+
+#ensure-grid-band-chart
+  _ _ ratio-multiplier: ,
+  current-price: mul(
+      uniswap-v3-twap-output-ratio(
+        distribution-token
+        reserve-token
+        0 0
+        twap-io-fee
+      ) 
+      ratio-multiplier
+  );
+  
+
+#ensure-grid-band-prod
+  grid-min-price grid-max-price ratio-multiplier: ,
+  current-price: mul(
+      uniswap-v3-twap-output-ratio(
```

Comment:
how is this a grid strat if you're just taking the oracle price for every trade?

why are you using a 0 0 twap instead of an exact quote word?